### PR TITLE
Add 3 new examples to Cookbook.

### PR DESCRIPTION
**Pull request by vfscalfani**
This pull request adds 3 new examples to the Cookbook (35, 36, and 37). Two from Taka @iwatobipen (thanks!) showing:

1. How to add a molecule drawing to a matplotlib plot (entry 35)
2. How to highlight molecule differences based on maximum common substructure (entry 36)

And I added a simple example with CoordGen library (entry 37) as this seems to come up often when trying to draw macrocycles. This should make it easier to share the Cookbook link for an example.

Thanks,

Vin

### CODE REVIEW

**Comment by greglandrum** [comment]
@vfscalfani : the test failures here are because apparently all of the doctests end up being run in the same python interpreter, so the `setPreferCoordgen()` call done in the new example propagatest to all tests run after that one.
Unfortunately this includes the "Getting Started" guide when run in CI.

There are two easy fixes I can think of:
1. add a line with `rdDepictor.SetPreferCoordGen(False)` at the bottom of `RDKitCB_37`
2. add the following at around line `38` of `GettingStartedInPython.rst`:
```
  from rdkit.Chem import rdDepictor
  rdDepictor.SetPreferCoordGen(False)
```

Either is ok with me, though I have a slight preference for the first solution if you don't think it makes things too confusing.

**Comment by greglandrum** [comment]
The comment mentions `CB35`, but I don't see it in the actual PR

**Comment by vfscalfani** [comment]
Hi @greglandrum , thanks for the detailed explanation. I went with the preferred option 2 and have pushed the changes. It appears that I also needed to do the same to the RDKit_Book.rst file too, to get my local tests to pass. Hopefully the CI tests pass now, I'll check back later.

For CB35, that is on lines 467-503 of the Cookbook document. Yeah, it's confusing because we number things sequentially, but that doesn't necessarily mean they are in order from top to bottom. I would like to move away from the numbering scheme soon, and come up with a better way to index/cite an entry (e.g., with an individual DOI)... I'll look at options. Thanks, Vin

**Comment by greglandrum** [comment]
> For CB35, that is on lines 467-503 of the Cookbook document. Yeah, it's confusing because we number things sequentially, but that doesn't necessarily mean they are in order from top to bottom. I would like to move away from the numbering scheme soon, and come up with a better way to index/cite an entry (e.g., with an individual DOI)... I'll look at options. Thanks, Vin

Ah, sorry, I had just skimmed over that as being part of 37 since I wasn't reviewing yet.. I'll review now

File: Docs/Book/Cookbook.rst [hunk 1]
```diff
@@ -238,6 +238,60 @@ Highlight a Substructure in a Molecule
...
+| **Original Source:** `<https://gist.github.com/iwatobipen/6d8708d8c77c615cfffbb89409be730d>`_
+| **Index ID#:** RDKitCB_36
+| **Summary:** Highlight molecule differences based on maximum common substructure
+
+.. testcode::
+
+   from rdkit import Chem
+   from rdkit.Chem import Draw
+   from rdkit.Chem.Draw import IPythonConsole
+   from rdkit.Chem import rdMMPA
```
Comment (greglandrum) [inline Docs/Book/Cookbook.rst:254]: I don't think this is used anywhere

**Review by greglandrum — CHANGES_REQUESTED** [review]
one minor change requested

File: Docs/Book/Cookbook.rst (hunk 1 above)
Comment (vfscalfani) [reply Docs/Book/Cookbook.rst]: Agreed, and now removed.

**Review by greglandrum — APPROVED** [review]

**Comment by greglandrum** [comment]
Thanks @vfscalfani !

**Pull request merged**